kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/runtime/functions/CeilingFunction.java

```
@@ -45,11 +46,14 @@ public FEELFnResult<BigDecimal> invoke(@ParameterName( "n" ) BigDecimal n, @Para
         if ( scale == null ) {
             return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "scale", "cannot be null"));
         }
+        Object scaleObj = NumberEvalHelper.coerceIntegerNumber(scale);
         // Based on Table 76: Semantics of numeric functions, the scale is in range −6111 .. 6176
-        if (scale.compareTo(BigDecimal.valueOf(-6111)) < 0 || scale.compareTo(BigDecimal.valueOf(6176)) > 0) {
-            return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "scale", "must be in range between -6111 to 6176."));
+        if (scaleObj instanceof Integer) {
+            int scaleInt = (Integer) scaleObj;
+            if (scaleInt < -6111 || scaleInt > 6176) {
+                return FEELFnResult.ofError(new InvalidParametersEvent(Severity.ERROR, "scale", "must be in range between -6111 to 6176."));
+            }
```

else case could be written here? (i.e., if scaleObj is not an Integer). currently, the flow silently skips validation.

---

Modified this. Please check